.github/workflows/bitcoin-tests.yml

```diff
@@ -35,15 +38,23 @@ jobs:
         run: |
           rm .dockerignore
           docker build -f ./.github/actions/bitcoin-int-tests/Dockerfile.generic.bitcoin-tests -t stacks-blockchain:integrations .
+
+      - name: Free Disk Space (Ubuntu)
+        uses: jlumbroso/free-disk-space@main
+        with:
+          tool-cache: true
+          docker-images: false
+
```

Comment:
Is this step necessary with the gzip change?

I'm pretty wary of using third-party actions.

Reply:
without that step, it's *really* close. 
without cleaning up packages on the host, we may hit this issue again. 

if this is a blocker, the only other thing we could (outside of manually removing packages, which is what this action does) is as j2p2 suggested: `docker system prune --force` before our image is built. 
there's about 3-4GB of images loaded by default on the runner that we don't use/need.

the free-disk-space workflow is pretty simple and straightforward: https://github.com/jlumbroso/free-disk-space/blob/main/action.yml  but you're correct in that it is a 3rd party action and they could add some logic there to do something we don't like (the same can be said for other 3rd party actions we use though). 

if you feel strongly about not using it,  there should be just enough space by a combination of removing unnecessary docker images and using gzip vs tar. the caveat is that if the base packages change to use more space, we'll have this issue again (however, hopefully by then we won't even need this compressed docker image).

Reply:
ping @kantai - any thoughts on my reply?

Reply:
It's simple enough that we could just vendor it if we're worried.  I don't think Docker has the equivalent of Cargo.lock; if they truly don't, then that'd at least de-risk supply chain poisoning.

Reply:
Unless there is a strong reason to **not** use this action, i'd say leave it in and let it be the temporary workaround it was meant to be. 
I'm also working on some methods to make this change moot; it is only meant to fix a very current and specific issue,  and should not be something we use long term.

@jcnelson you do raise an interesting idea along with @kantai though - if there are concerns about using 3rd party workflows, we should have a discussion about recreating them ourselves or vendoring them.

Reply:
Understood - I do want to highlight that it feels like a "not built here" condition. But, I can refactor this PR to address your concerns and we can discuss what to do about 3rd party actions at some other time. 

I'll have a commit doing what i suggested in the first reply to your comment later today. It should be enough for the short term as a workaround to this issue.